src/js/packages/@reactpy/client/src/components.tsx

```diff
@@ -125,23 +125,15 @@ function ScriptElement({ model }: { model: ReactPyVdom }) {
       (value): value is string => typeof value == "string",
     )[0];
 
-    let scriptElement: HTMLScriptElement;
-    if (model.attributes) {
-      scriptElement = document.createElement("script");
-      for (const [k, v] of Object.entries(model.attributes)) {
-        scriptElement.setAttribute(k, v);
-      }
-      if (scriptContent) {
-        scriptElement.appendChild(document.createTextNode(scriptContent));
-      }
-      ref.current.appendChild(scriptElement);
-    } else if (scriptContent) {
-      const scriptResult = eval(scriptContent);
-      if (typeof scriptResult == "function") {
-        return scriptResult();
-      }
+    const scriptElement: HTMLScriptElement = document.createElement("script");
+    for (const [k, v] of Object.entries(model.attributes || {})) {
+      scriptElement.setAttribute(k, v);
+    }
+    if (scriptContent) {
+      scriptElement.appendChild(document.createTextNode(scriptContent));
     }
-  }, [model.key, ref.current]);
+    ref.current.appendChild(scriptElement);
+  }, [model.key]);
```

Comment:
I'm testing out this implementation in https://github.com/reactive-python/reactpy/pull/1239 but it seems that it breaks script re-execution when the script `key` (or `content`) changes.

Reply:
That sounds like the issue I've seen with inputs. You'll change the key and value on the input but it sticks around. It's an issue if you have a list of objects with inputs, update the list, then redraw. The browser shows the old value and doesn't update it.

The workaround IIRC is to wrap it and change the key on what its wrapped in.

I was under the impression my code change to support reconnection was the cause but maybe its not? If that's the case, it sounds like its a bug that affects everything, not just scripts.

Reply:
Also, just to call this out, I'm not sure having script tags execute multiple times because they changed is necessarily a good practice. I can think of a couple scenarios where implementation changes such as how renders are sent to the client might create behavioral changes due to "in-between" states getting skipped. Seems like it should always be one script one execution.

Reply:
I could agree with that - Given that sequential renders that affect the same components are "optimized out" by `layout.py`, there is definitely a reality where things could get skipped.

I am also aware that `layout.py` has some bugs, so it's on my plate to rewrite that.

See issue:
- https://github.com/reactive-python/reactpy/issues/1198